### PR TITLE
Implement searching for tag keys for processes/products/workflows

### LINKED ISSUE #473: [Bug]: Change productTag query filter to tag
### Contact Details

wouter.huisman@surf.nl

### What happened?

In the V2 UI the query syntax uses in subscriptions table the tag to filter on certain product tags. However in the processes table (workflows/tasks), `productTag` is used. Please change the query filter of the processes endpoint to use `tag` instead.

### Version

2.0.3

### What python version are you seeing the problem on?

Python 3.11

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by dmarjenburgh**
Added support for filtering products/processes/workflows by `tag:***`

And a bunch of tests to cover it.

Closes #473 

NOTE:
Diff contains unrelated reformatting changes.

### CODE REVIEW

File: test/unit_tests/utils/test_search_query.py [hunk 1]
```diff
@@ -102,3 +104,28 @@ def test_query_kv_camelcasing():
...
+)
+def test_sqlalchemy_visitor(search_query, substrings):
+    class MyTable(BaseModel):
+        __tablename__ = "my_table"
+        id = Column(Integer, primary_key=True)
+        name = Column(String)
+
+    mappings = {"name": lambda node: MyTable.name.like(node[1].lower())}
+    stmt = select(MyTable)
+    stmt = create_sqlalchemy_select(stmt, search_query, mappings, MyTable, MyTable.id)
```
Comment (mrijk) [inline test/unit_tests/utils/test_search_query.py:127]: Every time you reuse a variable somewhere on this world a baby seal dies ;)

**Review by mrijk — APPROVED** [review]
LGTM

File: orchestrator/db/filters/process.py [hunk 2]
```diff
@@ -128,32 +130,31 @@ def workflow_name_filter(query: QueryType, value: str) -> QueryType:
         "istask": generic_bool_filter(ProcessTable.is_task),  # TODO: will be removed in 1.4
         "status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: will be removed in 1.4
         "last_status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: will be removed in 1.4
-        "tag": tag_filter,  # TODO: will be removed in 1.4
```
Comment (Mark90) [inline orchestrator/db/filters/process.py:131]: I forgot why this was marked as `TODO: will be removed in 1.4`. @tjeerddie do you remember?

File: test/unit_tests/utils/test_search_query.py (hunk 1 above)
Comment (Mark90) [reply test/unit_tests/utils/test_search_query.py:127]: Generally I agree but for query building I don't see much benefit in creating new variables that you'll never need

i.e.

```
stmt_start = select(MyTable)
stmt_with_select = create_sqlalchemy_select(stmt_start)
stmt_with_stuff = add_more_stuff(stmt_with_select)
stmt_with_more = even_more(stmt_with_stuff) if something else stmt_with_stuff
...
```

vs

```
stmt = select(MyTable)
stmt = create_sqlalchemy_select(stmt)
stmt = add_more_stuff(stmt)
if something:
  stmt = even_more(stmt)
```

Might just be an inherent problem with using an ORM to create dynamic queries

**Review by Mark90 — APPROVED** [review]

**Pull request merged**